File: app/src/main/java/me/clarius/sdk/cast/example/ProcedureSelectionFragment.java
package me.clarius.sdk.cast.example;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.Spinner;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.navigation.Navigation;

public class ProcedureSelectionFragment extends Fragment {

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_procedure_selection, container, false);

        Spinner spinner = view.findViewById(R.id.spinner_procedures);
        Button buttonStart = view.findViewById(R.id.button_start);

        // Set up the spinner with procedure names
        ArrayAdapter<CharSequence> adapter = ArrayAdapter.createFromResource(getContext(),
                R.array.procedures_array, android.R.layout.simple_spinner_item);
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        spinner.setAdapter(adapter);

        buttonStart.setOnClickListener(v -> {
            String selectedProcedure = spinner.getSelectedItem().toString();
            ProcedureSelectionFragmentDirections.ActionProcedureSelectionFragmentToSlidesFragment action =
                    ProcedureSelectionFragmentDirections.actionProcedureSelectionFragmentToSlidesFragment(selectedProcedure);
            Navigation.findNavController(view).navigate(action);
        });

        return view;
    }
}

File: app/src/main/java/me/clarius/sdk/cast/example/ProcedureSelectionFragment.kt
package me.clarius.sdk.cast.example

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.Spinner
import androidx.fragment.app.Fragment
import androidx.navigation.Navigation.findNavController

class ProcedureSelectionFragment : Fragment() {
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.fragment_procedure_selection, container, false)

        val spinner = view.findViewById<Spinner>(R.id.spinner_procedures)
        val buttonStart = view.findViewById<Button>(R.id.button_start)

        // Set up the spinner with procedure names
        val adapter = ArrayAdapter.createFromResource(
            requireContext(),
            R.array.procedures_array, android.R.layout.simple_spinner_item
        )
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        spinner.adapter = adapter

        buttonStart.setOnClickListener { v: View? ->
            val selectedProcedure = spinner.selectedItem.toString()
            val action =
                ProcedureSelectionFragmentDirections.actionProcedureSelectionFragmentToSlidesFragment(
                    selectedProcedure
                )
            findNavController(view).navigate(action)
        }

        return view
    }
}